dkg-gadget/src/worker.rs

```diff
@@ -850,6 +850,12 @@ where
 		debug!(target: "dkg_gadget::worker", "🕸️  Processing block notification for block {}", header.number());
 		*self.latest_header.write() = Some(header.clone());
 		debug!(target: "dkg_gadget::worker", "🕸️  Latest header is now: {:?}", header.number());
+		// Check if we should execute emergency Keygen Protocol.
+		if self.should_execute_emergency_keygen(header) {
+			debug!(target: "dkg_gadget::worker", "🕸️  Should execute emergency keygen");
+			self.handle_emergency_keygen(header);
```

Comment:
QQ : Will it be ok if a queued or current DKG is already in process and we trigger emergency keygen?

Reply:
There is handling already for that Edge case, basicly it will just ignore that call and return.
https://github.com/webb-tools/dkg-substrate/blob/4eaee85a2253e95a5c54dd29846bf2c298445d75/dkg-gadget/src/worker.rs#L794-L804